src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,375 @@
+    /// </param>
+    /// <returns>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </returns>
+    public static IEnumerable<(AddWindowReorderingSupportToSiblings ReorderingNode, Node Sibling)>
+        GetWindowReorderingAncestors(Node startingNode, int maxSearchDepth, Array<NodePath>? reorderingNodePaths)
+    {
+        if (reorderingNodePaths == null || reorderingNodePaths.Count == 0)
+        {
+            return GetWindowReorderingAncestors(startingNode, maxSearchDepth);
+        }
+
+        return GetWindowReorderingAncestors(startingNode, reorderingNodePaths);
+    }
+
+    public override void _ExitTree()
+    {
+        connectionsEstablished = false;
+        windowReorderingAncestors.Clear();
+        connectedWindows.Clear();
+        connectedSiblings.Clear();
+        topSibling = null;
+    }
+
+    public void ConnectWindow(CustomDialog window, Node topNode, bool recursive)
+    {
+        // Make sure to be connected to other window reordering nodes before trying to connect the window
+        Setup();
+
+        if (recursive)
+        {
+            // Connection is recursive, connect to other window reordering nodes
+            foreach (var ancestor in windowReorderingAncestors)
+            {
+                ancestor.ReorderingNode.ConnectWindow(window, ancestor.Sibling, recursive);
+            }
+        }
+
+        if (window.IsConnected(nameof(Dragged), this, nameof(OnWindowReorder)))
+        {
+            // This window is already connected here
+            return;
+        }
+
+        window.Connect(nameof(Dragged), this, nameof(OnWindowReorder));
+
+        connectedWindows.Add(window, topNode);
+        connectedSiblings.Add(topNode);
+
+        // Update top sibling
+        if (topSibling == null || topSibling.GetIndex() < topNode.GetIndex())
+            topSibling = topNode;
+    }
+
+    public void DisconnectWindow(CustomDialog window, bool recursive)
+    {
+        if (recursive)
+        {
+            // Connection was recursive, disconnect from other window reordering nodes
+            foreach (var ancestor in windowReorderingAncestors)
+            {
+                ancestor.ReorderingNode.DisconnectWindow(window, recursive);
+            }
+        }
+
+        if (!window.IsConnected(nameof(Dragged), this, nameof(OnWindowReorder)))
+        {
+            // This window has already been disconnected from here
+            return;
+        }
+
+        window.Disconnect(nameof(Dragged), this, nameof(OnWindowReorder));
+
+        var windowSibling = connectedWindows[window];
+        connectedWindows.Remove(window);
+
+        // Find if another connected window has the same sibling
+        bool foundSibling = false;
+        foreach (var sibling in connectedSiblings)
+        {
+            if (sibling == windowSibling)
+            {
+                foundSibling = true;
+                break;
+            }
+        }
```

Comment:
```suggestion
        bool foundSibling = connectedSiblings.Contains(windowSibling);
```

Isn't that the equivalent code? No need to manually search through a list.